### PR TITLE
Created EDXAPP_ variables for all defined FEATURES.

### PR DESCRIPTION
**Pull request by misilot**
This allows for creating/updating the variables without having to copy
the whole FEATURES variable into EDXAPP_ENV_EXTRA

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [X] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @misilot! I've created [OSPR-4944](https://openedx.atlassian.net/browse/OSPR-4944) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by morenol** [comment]
@misilot this looks fine to me, what do you think if we create an `EDXAPP_FEATURES_DEFAULT` variable equals to the actual value of `EDXAPP_FEATURES`. And then a `EDXAPP_FEATURES_EXTRA` variable which by default is an empty dict. And then we define `EDXAPP_FEATURES` as:

 ```yaml
EDXAPP_FEATURES: {{ EDXAPP_FEATURES_DEFAULT | combine(EDXAPP_FEATURES_EXTRA) }}
```

So we can also add new entries to the EDXAPP_FEATURES by adding elements to the `EDXAPP_FEATURES_EXTRA` variable.
Similar to what is done in: https://github.com/edx/configuration/blob/71db80f13555a1f912368f50306961d0cc787819/playbooks/roles/edx_django_service/defaults/main.yml#L45

**Comment by misilot** [comment]
@morenol that actually makes more sense. I hadn't utilized the combine() filter before, so I will work on updating the PR to utilize your recommendation.

I had run into a situation when I tried to utilize the EDXAPP_ENV_EXTRA with FEATURES as a key it would blow away everything else except for what is in the the new FEATURES key.

**Comment by morenol** [comment]
> @morenol that actually makes more sense. I hadn't utilized the combine() filter before, so I will work on updating the PR to utilize your recommendation.

I think that we can keep the changes that you already made and in top of that add the thing that I suggested.

> I had run into a situation when I tried to utilize the EDXAPP_ENV_EXTRA with FEATURES as a key it would blow away everything else except for what is in the the new FEATURES key.

I guess that it is happening because in the .yml files the configs are updated at this way.
https://github.com/edx/configuration/blob/ecef501edf17c8222b89600d8d7466a7c8410f2c/playbooks/roles/edxapp/templates/lms.yml.j2#L2

**Comment by natabene** [comment]
@misilot Thank you for the contribution! When you have a chance, could you please send in a signed Contributor Agreement mentioned in the earlier comment? You only need to do this once and we can then review all your future code contributions. Thanks

**Comment by misilot** [comment]
@natabene, I am attempting to find out if I can sign it. The notes in the message from the bot mention "indicated your institutional affiliation" is there an option for institutional affiliations to sign? If so is there more information in this regard?

**Comment by natabene** [comment]
@misilot Yes, absolutely. If you are part of an institution or a company, that may be interested in signing an entity agreement, please email your request to legal@edx.org, and they will guide you through the process.  Individual agreement can be signed if the code you plan to contribute was done by you on our own time and is not affiliated with any other party or company. Entity agreement is needed if you are contributing on behalf of your institution and you worked on your contribution as a part of your work project or assignment. Please request more details with our Legal team (legal@edx.org) if you need to.

**Comment by misilot** [comment]
Thanks @natabene I have forwarded that on to our Legal team, so hopefully they will reach out soon.

**Comment by fghaas** [comment]
@misilot Just as a heads up, I have PR #5887 open which adds a toggle for another defined feature, `ENABLE_BULK_ENROLLMENT_VIEW`. If you'd like to add that feature flag to your PR, I could close mine.

**Comment by misilot** [comment]
@fghaas once I hear if I can commit any code to the project I can cherrypick your change into this commit. However, until I hear when a contributor agreement has been signed, I would recommend leaving yours open as is.

**Comment by fghaas** [comment]
@misilot Sure, that works for me!

**Comment by natabene** [comment]
@misilot Thank you for joining the Open edX community! We have added your user to our records and can now review your code going forward.

**Comment by natabene** [comment]
@fredsmith Can you give this a look when you have a chance?

**Review by fredsmith — APPROVED** [review]
@misilot can you rebase your change?  I'll merge it asap.

**Comment by misilot** [comment]
Thanks @fredsmith, I will work on doing that. What are your thoughts of my cherry picking #5887 into my PR as requested by @fghaas

**Comment by misilot** [comment]
> 
> I think that we can keep the changes that you already made and in top of that add the thing that I suggested.
> 
@morenol I have added the changes you suggested.

File: playbooks/roles/edxapp/defaults/main.yml [hunk 1]
```diff
@@ -315,9 +329,13 @@ EDXAPP_FEATURES:
   ENABLE_READING_FROM_MULTIPLE_HISTORY_TABLES: "{{ EDXAPP_ENABLE_READING_FROM_MULTIPLE_HISTORY_TABLES }}"
   SHOW_HEADER_LANGUAGE_SELECTOR: "{{ EDXAPP_SHOW_HEADER_LANGUAGE_SELECTOR }}"
   SHOW_FOOTER_LANGUAGE_SELECTOR: "{{ EDXAPP_SHOW_FOOTER_LANGUAGE_SELECTOR }}"
-  ENABLE_ENROLLMENT_RESET: false
+  ENABLE_ENROLLMENT_RESET: "{{ EDXAPP_ENABLE_ENROLLMENT_RESET }}"
   ENABLE_EXPORT_GIT: "{{ EDXAPP_ENABLE_EXPORT_GIT }}"
 
+EDXAPP_FEATURES_EXTRA: {}
+
+EDXAPP_FEATURES: {{ EDXAPP_FEATURES_DEFAULT | combine(EDXAPP_FEATURES_EXTRA) }}
```
Comment (morenol) [inline playbooks/roles/edxapp/defaults/main.yml]: Hi, @misilot  I think that this needs to be inside quotes

Comment (misilot) [reply playbooks/roles/edxapp/defaults/main.yml]: Thanks @morenol

**Review by morenol — APPROVED** [review]

**Comment by misilot** [comment]
@fredsmith is there anything else you need for this?

**Review by coryleeio — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@misilot 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/5992&entry.1484655318=https://github.com/misilot&entry.752974735=2020-09-03+21:43&entry.1917517419=2020-10-15+16:32&entry.2133058324=Yes) so we can improve your experience in the future.

**Review by adzuci — COMMENTED** [review]
:shipit:

**Comment by coryleeio** [comment]
Looks good @misilot Thank you!

**Comment by misilot** [comment]
Thank you!

**Pull request merged**